samples/csharp_dotnetcore/21.corebot-app-insights/Dialogs/DateResolverDialog.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Recognizers.Text.DataTypes.TimexExpression;
+
+namespace Microsoft.BotBuilderSamples.Dialogs
+{
+    public class DateResolverDialog : CancelAndHelpDialog
+    {
+        public DateResolverDialog(string id = null)
+            : base(id ?? nameof(DateResolverDialog))
+        {
+            AddDialog(new DateTimePrompt(nameof(DateTimePrompt), DateTimePromptValidator));
```

Comment:
Are these produce application insights telemetry? if not, we need to add telemetry here as well.

Reply:
The telemetry client is set when added to the DialogSet